libraries/render-utils/src/MeshPartPayload.cpp

```diff
@@ -389,13 +129,76 @@ void ModelMeshPartPayload::updateClusterBuffer(const std::vector<Model::Transfor
     }
 }
 
-void ModelMeshPartPayload::updateTransformForSkinnedMesh(const Transform& renderTransform, const Transform& boundTransform) {
-    _transform = renderTransform;
-    _worldBound = _adjustedLocalBound;
-    _worldBound.transform(boundTransform);
+void ModelMeshPartPayload::computeAdjustedLocalBound(const std::vector<glm::mat4>& clusterMatrices) {
+    _adjustedLocalBound = _localBound;
+    if (clusterMatrices.size() > 0) {
+        _adjustedLocalBound.transform(clusterMatrices.back());
+
+        for (int i = 0; i < (int)clusterMatrices.size() - 1; ++i) {
+            AABox clusterBound = _localBound;
+            clusterBound.transform(clusterMatrices[i]);
+            _adjustedLocalBound += clusterBound;
+        }
+    }
+}
+
+void ModelMeshPartPayload::computeAdjustedLocalBound(const std::vector<Model::TransformDualQuaternion>& clusterDualQuaternions) {
+    _adjustedLocalBound = _localBound;
+    if (clusterDualQuaternions.size() > 0) {
+        Transform rootTransform(clusterDualQuaternions.back().getRotation(),
+                                clusterDualQuaternions.back().getScale(),
+                                clusterDualQuaternions.back().getTranslation());
+        _adjustedLocalBound.transform(rootTransform);
+
+        for (int i = 0; i < (int)clusterDualQuaternions.size() - 1; ++i) {
+            AABox clusterBound = _localBound;
+            Transform transform(clusterDualQuaternions[i].getRotation(),
+                                clusterDualQuaternions[i].getScale(),
+                                clusterDualQuaternions[i].getTranslation());
+            clusterBound.transform(transform);
+            _adjustedLocalBound += clusterBound;
+        }
+    }
+}
+
+void ModelMeshPartPayload::updateTransformForSkinnedMesh(const Transform& modelTransform, const Model::MeshState& meshState, bool useDualQuaternionSkinning) {
+    _localTransform = Transform();
+    if (useDualQuaternionSkinning) {
+        if (meshState.clusterDualQuaternions.size() == 1 || meshState.clusterDualQuaternions.size() == 2) {
+            const auto& dq = meshState.clusterDualQuaternions[0];
+            _localTransform = Transform(dq.getRotation(),
+                                       dq.getScale(),
+                                       dq.getTranslation());
+        }
+    } else {
+        if (meshState.clusterMatrices.size() == 1 || meshState.clusterMatrices.size() == 2) {
+            _localTransform = Transform(meshState.clusterMatrices[0]);
+        }
+    }
+
+    _parentTransform = modelTransform;
```

Comment:
in order to support billboarding, we keep track of the local transform of this part separately from from the "parentTransform" (transform of the model).  we can then use the parentTransform for billboarding and only apply the local transform after

Reply:
Seems like it's working as expect both with the test mentioned above and bringing in my own model.  As long as the mesh is setup properly in your 3d application, this works very well.  Nice work.